test/sequential/test-child-process-emfile.js

```diff
@@ -1,30 +1,46 @@
 'use strict';
-var common = require('../common');
-var assert = require('assert');
-var spawn = require('child_process').spawn;
-var fs = require('fs');
+const common = require('../common');
+const assert = require('assert');
+const child_process = require('child_process');
+const fs = require('fs');
 
 if (common.isWindows) {
   console.log('1..0 # Skipped: no RLIMIT_NOFILE on Windows');
   return;
 }
 
-var openFds = [];
+const ulimit = Number(child_process.execSync('ulimit -n'));
```

Comment:
@jbergstroem `ulimit -Hn <integer>` works without super-user privs for me if and only if I first set the soft limit as low or lower than what I'm trying to set the hard limit to. (In retrospect: Duh!) And `sudo` doesn't fix it. I thought it did because I was testing in different shell instances and must have had the soft limit already set appropriately in one of them.

TL;DR: You were right to be surprised (skeptical?) of my results.

The @bnoordhuis plan of setting with `-n` and querying with `-Hn` makes the most sense to me.